client/ayon_core/pipeline/load/plugins.py

```diff
@@ -433,6 +440,187 @@ def wrapped_method(self, *args, **kwargs):
+
+    """
+    order: int = 100
+    settings_category: str = None
+    settings_name: str = None
+
+    def __init__(
+        self,
+        load_context: LoadContext,
+        project_settings: dict[str, Any],
+    ) -> None:
+        self._log = None
+        self._load_context = load_context
+        self.apply_settings(project_settings)
+
+    @property
+    @abstractmethod
+    def identifier(self) -> str:
+        """Plugin identifier."""
+        pass
+
+    @property
+    def log(self) -> logging.Logger:
+        if self._log is None:
+            self._log = Logger.get_logger(self.__class__.__name__)
+        return self._log
+
+    def apply_settings(self, project_settings: dict[str, Any]) -> None:
+        """Apply project settings to the plugin.
+
+        Args:
+            project_settings (dict[str, Any]): Project settings.
+
+        """
+        cls_name = self.__class__.__name__
+        settings_name = self.settings_name or cls_name
+
+        settings = project_settings
+        for key in (
+            self.settings_category,
+            "load",
+            settings_name,
+        ):
+            settings = settings.get(key)
+            if settings is None:
+                self.log.debug(f"No settings found for {cls_name}")
+                return
+
+        for key, value in settings.items():
+            # Log out attributes that are not defined on plugin object
+            # - those may be potential dangerous typos in settings
+            if not hasattr(self, key):
+                self.log.debug(
+                    "Applying settings to unknown attribute"
+                    f" '{key}' on '{cls_name}'."
+                )
+            setattr(self, key, value)
+
+    @abstractmethod
+    def collect_containers(self) -> None:
+        """Collect containers from the current workfile.
+
+        This method is called by LoadContext on initialization and on reset.
+        """
+        pass
+
+    @abstractmethod
+    def is_representation_compatible(
+        self,
+        representation_context: RepresentationContext,
+    ) -> bool:
+        """Check if representation is compatible with the plugin.
+
+        Args:
+            representation_context (RepresentationContext): Representation
+                context.
+
+        Returns:
+            bool: True if compatible, False otherwise.
+
+        """
+        pass
+
+    @abstractmethod
+    def load_representations(
+        self,
+        representation_contexts: list[RepresentationContext],
+    ) -> list[ContainerItem]:
+        """Load representations.
+
+        Method still has to call 'add_containers' method on 'LoadContext'.
+
+        Args:
+            representation_contexts (list[RepresentationContext]): List of
+                representation contexts.
+
+        Returns:
+            list[ContainerItem]: List of loaded containers.
+
+        """
+        pass
+
+    @abstractmethod
+    def change_representations(
```

Comment:
The 'change' terminology sounds very new to the load API that existed. I wonder if this should be `update_representation` instead? :/

Reply:
You can change to different (lower) version or switch container of other loader.

EDITED:
I see that I did add `switch_containers` but it would make more sense to actually use this method instead as you might also want to change the representation at the same time.